src/main/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/kafka/KafkaRecordConsumer.kt

```diff
@@ -118,16 +128,31 @@ abstract class KafkaRecordConsumer<T, R>(
 		}
 	}
 
+	internal open fun createKafkaConsumer(props: Properties) = KafkaConsumer<Key, T>(props)
+
+	open fun shouldStop(newRecords: List<ConsumerRecord<Key, T>>) =
+		newRecords.isNotEmpty() && newRecords.last().timestamp() > startTime
```

Comment:
Vet ikke om jeg har fått med meg full context her. Variabelen startTime settes når klassen instansieres (ved start av gjennomgang av input strømmen). Sjekker du  `newRecords.last().timestamp() > startTime` for å finne records som er lagt på input strømmen etter oppstart?

Reply:
Riktig. Hvis vi ser en melding som lagts på topicen etter at vi startet å lese, så skal vi returnere.

Reply:
Grunden til det er at dette brukes av bootstrappinga ved oppstart. Vi kan ellers hamne i en situasjon då en pod producerer til topic og den andre konsumerer den forstes records uten at noensin bli ferdig.